src/storage/addtoCard.js

```diff
@@ -9,58 +9,66 @@ const AddToCardCtx = createContext({
   removingProduct: (product) => {},
   addingProduct: (product) => {},
   isInCard: () => {},
+  productCountReSet: (number) => {},
 });
 
 //context provider component
 const AddtoCardCtxProvider = ({ children }) => {
-  //default value useEffect
-  useEffect(() => {
-    const cardsProducts = getLocalStorger();
-    setAddedProduct(cardsProducts);
-  }, []);
+  //function of getting from localstorage
+  const getLocalStorge = () => {
+    let cardsProducts = localStorage.getItem("cardsProducts");
+    try {
+      if (cardsProducts) {
+        return JSON.parse(localStorage.getItem("cardsProducts"));
+      } else {
+        return [];
+      }
+    } catch (err) {
+      console.log(err);
+    }
+  };
 
+  // states
   const [productCount, setProductCount] = useState(0);
-  const [addedProduct, setAddedProduct] = useState([]);
+  const [addedProduct, setAddedProduct] = useState(getLocalStorge);
```

Comment:
in function mitoone omoomi tar bashe va maslan esmesh, `readArrayFromLocalStorage` bashe ke, ye key migire va uno az local storage mikhone, bad to in file va un yeki file estefade beshe